tests/multi_ptr/multi_ptr_convert_assignment_ops.h

```diff
@@ -0,0 +1,198 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides code for multi_ptr convert assignment operators
+//
+*******************************************************************************/
+
+#ifndef __SYCLCTS_TESTS_MULTI_PTR_CONVERT_ASSIGN_OPS_H
+#define __SYCLCTS_TESTS_MULTI_PTR_CONVERT_ASSIGN_OPS_H
+
+#include "../common/common.h"
+
+#include "../common/section_name_builder.h"
+#include "multi_ptr_common.h"
+
+namespace multi_ptr_convert_assignment_ops {
+
+constexpr int expected_val = 42;
+
+template <typename T, typename SrcAddrSpaceT, typename SrcIsDecorated,
+          typename DstIsDecorated>
+class run_convert_assignment_operators_tests {
+  static constexpr sycl::access::address_space src_space = SrcAddrSpaceT::value;
+  static constexpr sycl::access::decorated src_decorated =
+      SrcIsDecorated::value;
+  static constexpr sycl::access::decorated dst_decorated =
+      DstIsDecorated::value;
+  using src_multi_ptr_t = sycl::multi_ptr<T, src_space, src_decorated>;
+  using dst_multi_ptr_t =
+      sycl::multi_ptr<T, sycl::access::address_space::generic_space,
+                      src_decorated>;
+
+ public:
+  void operator()(const std::string &type_name,
+                  const std::string &src_address_space_name,
+                  const std::string &src_is_decorated_name,
+                  const std::string &dst_is_decorated_name) {
+    auto queue = sycl_cts::util::get_cts_object::queue();
+    T value = user_def_types::get_init_value_helper<T>(expected_val);
+    auto r = sycl::range(1);
+    SECTION(
+        section_name(
+            "Check &operator=(const multi_ptr<value_type, ASP, IsDecorated>&)")
+            .with("T", type_name)
+            .with("src address_space", src_address_space_name)
+            .with("src decorated", src_is_decorated_name)
+            .with("dst address_space", "access::address_space::generic_space")
+            .with("dst decorated", dst_is_decorated_name)
+            .create()) {
+      bool res = false;
+      {
+        sycl::buffer<bool> res_buf(&res, r);
+        sycl::buffer<T> val_buffer(&value, r);
+        queue.submit([&](sycl::handler &cgh) {
+          auto res_acc =
+              res_buf.template get_access<sycl::access_mode::write>(cgh);
+          auto acc_for_mptr =
+              val_buffer.template get_access<sycl::access_mode::read>(cgh);
+
+          if constexpr (src_space ==
+                        sycl::access::address_space::global_space) {
+            auto val_acc =
+                val_buffer.template get_access<sycl::access_mode::read>(cgh);
+            cgh.single_task([=] {
+              const src_multi_ptr_t mptr_in(acc_for_mptr);
+              dst_multi_ptr_t mptr_out;
+
+              mptr_out = mptr_in;
```

Comment:
> Does this PR assume that it's possible to cast multi_ptr of any address space?

This PR assumes that we can cast `multi_ptr` of any `access::address_space` to `multi_ptr` with `access::address_space::generic_space` 

Is this what you meant in your question?

Reply:
> Does this PR assume that it's possible to cast `multi_ptr` of any address space? Because I don't think that's quite the case, at least not in the reduced feature set where generics might not be available.

Is there a query to test which feature set is supported by the implementation or it should be a CTS configuration?
At the moment, CTS covers full feature set and I think we can address this issue separately. I see you suggested this approach here: https://github.com/KhronosGroup/SYCL-CTS/pull/350#discussion_r911159471.
I  suggest we open an issue to enable support for reduced feature sets and merge this as is. @ProGTX, does it work for you?

Reply:
That works for me, I think we're going to need a more comprehensive approach for dealing with the reduced feature set.

Reply:
Great. I'll open an issue to add support for reduced feature set to SYCL-CTS and merge this change-set.